clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -620,7 +620,12 @@ public synchronized Optional<Integer> preferredReadReplica(TopicPartition tp, lo
      * @return the removed preferred read replica if set, None otherwise.
      */
     public synchronized Optional<Integer> clearPreferredReadReplica(TopicPartition tp) {
-        return assignedState(tp).clearPreferredReadReplica();
+        final TopicPartitionState topicPartitionState = assignedStateOrNull(tp);
+        if (topicPartitionState == null) {
+            return Optional.empty();
+        } else {
```

Comment:
The previous condition was returned, so you don't need the else.

Reply:
Sorry. I did not see your comment before merging. I agree that we could remove it but it is also ok with it. Other methods in this class are structured like this.